packages/fether-react/src/Send/TxForm/TxForm.js

```diff
@@ -35,28 +35,58 @@ const MIN_GAS_PRICE = 3; // Safelow gas price from GasStation, in Gwei
 @withEthBalance // ETH balance
 @observer
 class Send extends Component {
+  state = {
+    maxSelected: false
+  };
   handleSubmit = values => {
     const { accountAddress, history, sendStore, token } = this.props;
 
     sendStore.setTx(values);
     history.push(`/send/${token.address}/from/${accountAddress}/signer`);
   };
 
-  decorator = createDecorator({
-    field: /to|amount/, // when the value of these fields change...
-    updates: {
-      // ...set field "gas"
-      gas: (value, allValues) => {
-        const { parityStore, token } = this.props;
-
-        if (this.preValidate(allValues) === true) {
-          return estimateGas(allValues, token, parityStore.api);
-        } else {
-          return null;
+  decorator = createDecorator(
+    {
+      field: /to|amount/, // when the value of these fields change...
+      updates: {
+        // ...set field "gas"
+        gas: (value, allValues) => {
+          const { parityStore, token } = this.props;
+          if (this.preValidate(allValues) === true) {
+            return estimateGas(allValues, token, parityStore.api);
+          } else {
+            return null;
+          }
+        }
+      }
+    },
+    {
+      field: /gas|gasPrice/, // when the value of these fields change...
+      updates: {
+        // ...set field "gas"
+        amount: (value, allValues) => {
```

Comment:
imo, from what I understood from final form:
- if you want to create a new field from other fields, use calculate
- if you want to mutate an existing field, use mutators

Here, using calculate to mutate an existing field, seems to work, but seems a bit hacky to me. Would it be possible to add a mutator when we slide the gas slider?